service/src/parentchain_block_syncer.rs

```diff
@@ -0,0 +1,79 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use itp_api_client_extensions::ChainApi;
+use itp_enclave_api::{enclave_base::EnclaveBase, sidechain::Sidechain};
+use itp_types::SignedBlock;
+use log::{error, trace};
+use my_node_runtime::Header;
+use std::cmp::min;
+
+const BLOCK_SYNC_BATCH_SIZE: u32 = 1000;
+
+pub(crate) struct ParentchainBlockSyncer<Api: ChainApi, EnclaveApi: EnclaveBase + Sidechain> {
+	node_api: Api,
+	enclave_api: EnclaveApi,
+}
+
+impl<Api, EnclaveApi> ParentchainBlockSyncer<Api, EnclaveApi>
+where
+	Api: ChainApi,
+	EnclaveApi: EnclaveBase + Sidechain,
```

Comment:
Is EnclaveBase trait still necessary here?

Reply:
no it is not :)